paimon-core/src/main/java/org/apache/paimon/AbstractFileStore.java

```diff
@@ -160,7 +161,8 @@ public FileStoreCommitImpl newCommit(String commitUser) {
                 options.manifestFullCompactionThresholdSize(),
                 options.manifestMergeMinCount(),
                 partitionType.getFieldCount() > 0 && options.dynamicPartitionOverwrite(),
-                newKeyComparator());
+                newKeyComparator(),
+                new CommitMetrics(pathFactory()));
```

Comment:
Why create `CommitMetrics` here? `CommitMetrics` is only used inside `FileStoreCommitImpl`. Users do not need to know its existence.